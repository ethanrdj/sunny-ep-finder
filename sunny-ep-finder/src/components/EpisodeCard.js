import React from "react";

import "../styles/EpisodeCard.css";

const EpisodeCard = ({
  series,
  episode,
  title,
  date,
  directedBy,
  writtenBy,
  description,
  handleRequest,
  setFilteredSeries,
  foundEp,
}) => {
  const handleChange = (event) => {
    setFilteredSeries(event.target.value);
  };
  return (
    <div className="EpisodeCard">
      <button className="button-filter" onClick={() => handleRequest()}>
        IASIP
      </button>
      <label className="button-filter" htmlFor="filter">
        Series:
      </label>
      <select
        name="filter"
        id="filter"
        className="button-filter"
        onChange={handleChange}
      >
        <option>All</option>
        <option>1</option>
        <option>2</option>
        <option>3</option>
        <option>4</option>
        <option>5</option>
        <option>6</option>
        <option>7</option>
        <option>8</option>
        <option>9</option>
        <option>10</option>
        <option>11</option>
        <option>12</option>
        <option>13</option>
        <option>14</option>
      </select>
      {foundEp.length === undefined && (
        <div className="result">
          <p className="title">
            <strong>{title}</strong>
          </p>
          <p className="series-ep">
            <strong>S</strong>
            {series}&nbsp;
            <strong>E</strong>
            {episode}
          </p>
          <p className="series-ep date">{date}</p>
          <br></br>
          <p className="dire-writ directedBy">
            <strong>Directed by:</strong>&nbsp;&nbsp;
            {directedBy}
          </p>
          <p className="dire-writ writtenBy">
            <strong>Written By:</strong>&nbsp;&nbsp;
            {writtenBy}
          </p>
          <p className="description">{description}</p>
        </div>
      )}
    </div>
  );
};

export default EpisodeCard;
