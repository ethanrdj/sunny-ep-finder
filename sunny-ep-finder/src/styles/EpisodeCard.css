@import url("https://fonts.googleapis.com/css2?family=Acme&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Raleway:wght@500&display=swap");
@import url("https://fonts.googleapis.com/css2?family=DM+Sans:ital@1&family=Fira+Sans+Condensed:ital@1&display=swap");

.EpisodeCard button {
  width: 45%;
  font-family: "Acme", sans-serif;
  font-size: 1.4em;
  background-color: #f3d111;
  border-radius: 5%;
  border: none;
  box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.2), 0 3px 10px 0 rgba(0, 0, 0, 0.19);
  cursor: pointer;
}

.EpisodeCard label {
  margin-left: 5%;
  font-family: "Acme", sans-serif;
  font-size: 1.4em;
}

.EpisodeCard select {
  margin-left: 1%;
  width: 15%;
  padding: 1%;
  font-size: 1.1em;
  border-radius: 5px;
  font-family: "Acme", sans-serif;
  cursor: pointer;
}

.result {
  margin: 0 auto;
  margin-top: 4%;
  font-family: "Acme", sans-serif;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 93%;
  background-color: #dcdcdc;
  border: 2px black solid;
  border-radius: 5%;
  box-shadow: 0 6px 8px 0 rgba(0, 0, 0, 0.2), 0 3px 18px 0 rgba(0, 0, 0, 0.19);
}

.title {
  font-family: "sunny";
  width: 90%;
  padding: 3%;
  font-size: 1.5em;
  margin-bottom: -6px;
  border-radius: 5px;
  background-color: black;
  color: white;
}

.series-ep {
  display: inline-block;
  margin-top: 3%;
  margin-bottom: 0;
  font-size: 1.3em;
}

.date {
  margin-top: 0;
}

.dire-writ {
  font-size: 1em;
  margin-top: -8%;
  padding: 4%;
}

.directedBy {
  margin-bottom: 0;
}

.description {
  font-family: "Acme", sans-serif;
  width: 90%;
  font-size: 1.5em;
  margin-top: -4%;
}

@media (min-width: 750px) {
  .dire-writ {
    padding-bottom: 5%;
  }

  .directedBy {
    padding-top: 6%;
  }
}

@media (min-width: 1400px) {
  .directedBy {
    padding-top: 7%;
    padding-bottom: 4%;
  }

  .EpisodeCard {
    width: 75%;
    margin: 0 auto;
  }
}
