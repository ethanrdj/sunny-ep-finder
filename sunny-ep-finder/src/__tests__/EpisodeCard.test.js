import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import EpisodeCard from "../components/EpisodeCard";

describe("EpisodeCard", () => {
  const episode = {
    series: 1,
    episode: 1,
    title: "test title",
    date: "test date",
    directedBy: "Someone",
    writtenBy: "someone else",
    description: "Its a TV show",
  };

  it("renders the correct props on the page", () => {
    const { asFragment } = render(
      <EpisodeCard episode={episode} foundEp="1" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
