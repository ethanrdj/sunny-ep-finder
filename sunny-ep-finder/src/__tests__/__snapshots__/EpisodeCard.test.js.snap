// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCard renders the correct props on the page 1`] = `
<DocumentFragment>
  <div
    class="EpisodeCard"
  >
    <button
      class="button-filter"
    >
      IASIP
    </button>
    <label
      class="button-filter"
      for="filter"
    >
      Series:
    </label>
    <select
      class="button-filter"
      id="filter"
      name="filter"
    >
      <option>
        All
      </option>
      <option>
        1
      </option>
      <option>
        2
      </option>
      <option>
        3
      </option>
      <option>
        4
      </option>
      <option>
        5
      </option>
      <option>
        6
      </option>
      <option>
        7
      </option>
      <option>
        8
      </option>
      <option>
        9
      </option>
      <option>
        10
      </option>
      <option>
        11
      </option>
      <option>
        12
      </option>
      <option>
        13
      </option>
      <option>
        14
      </option>
    </select>
  </div>
</DocumentFragment>
`;
